mycore-base/src/main/java/org/mycore/frontend/cli/MCRCommandUtils.java

```diff
@@ -0,0 +1,169 @@
+     * @return A (parallel) stream with found IDs. (may be empty if none were found)
+     * @throws MCRUsageException
+     *     If no type was given or the type could not be found.
+     *     Not thrown if type exists but has no values.
+     */
+    public static final Stream<String> getIdsForType(final String type) throws MCRUsageException {
+        if (type == null || type.length() == 0) {
+            throw new MCRUsageException("Type required to enumerate IDs!");
+        }
+        if (MCRConfiguration2.getString("MCR.Metadata.Type." + type).isEmpty()) {
+            throw new MCRUsageException(
+                "MCR.Metadata.Type." + type + " not set!");
+        }
+        List<String> idList;
+        try {
+            idList = MCRXMLMetadataManager.instance().listIDsOfType(type);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new MCRUsageException("Requested type " + type + " not found!", e);
+        }
+        if (idList.size() == 0) {
+            LOGGER.warn("No IDs found for type {}.", type);
+        }
+        return idList.parallelStream();
+    }
+
+    /**
+     * Get a stream of MCRObjectIDs found for a project &amp; type combination.
+     * 
+     * @param project
+     *     The project to look in the store for.
+     * @param type
+     *     The type to look in the store for.
+     * @return A (parallel) stream with found IDs. (may be empty if none were found)
+     * @throws MCRUsageException
+     *     If no project was given, no type was given or the base (${project}_${base}) could not be found.
+     *     Not thrown if the base exists but has no values.
+     */
+    public static final Stream<String> getIdsForProjectAndType(final String project, final String type)
+        throws MCRUsageException {
+        if (project == null || project.length() == 0) {
+            throw new MCRUsageException("Project required to enumerate IDs!");
+        }
+        if (type == null || type.length() == 0) {
+            throw new MCRUsageException("Type required to enumerate IDs!");
+        }
+        return getIdsForBaseId(project + "_" + type);
+    }
+
+    /**
+     * Get a stream of MCRObjectIDs found for a base.
+     * 
+     * @param base
+     *     The base to look in the store for.
+     * @return A (parallel) stream with found IDs. (may be empty if none were found)
+     * @throws MCRUsageException
+     *     If no base was given or the base could not be found.
+     *     Not thrown if the base exists but has no values.
+     */
+    public static final Stream<String> getIdsForBaseId(final String base) throws MCRUsageException {
+        if (base == null || base.length() == 0) {
+            throw new MCRUsageException("Base ID ({project}_{type}) required to enumerate IDs!");
+        }
+        List<String> idList;
+        try {
+            idList = MCRXMLMetadataManager.instance().listIDsForBase(base);
+        } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
do not catch RuntimeException here, code may change and will leave this code dead here.